internal/gcsx/temp_file.go

```diff
@@ -108,6 +113,38 @@ func NewTempFile(
 	return
 }
 
+// NewCacheFile creates a temp file whose initial contents are given by the
+// supplied reader. dir is a directory on whose file system the file will live,
+// or the system default temporary location if empty.
+func NewCacheFile(
```

Comment:
The package "gcsx" means "extended gcs". It contains the structs for communicating with the GCS API.

This NewCacheFile function is specifically used by the cache. How a cache file is written to the disk doesn't matter to the GCS communication. So, it should live in the contentcache package instead.

Reply:
@lezh  I thought we wanted GCS API to interact with cached files that instead of the TempFile.  This allows users to modify files and "writethrough" the changes to the cache before pushing the changes to GCS.

Reply:
We should clarify the responsibilities of each package. Depending on your design, the scope of the `fs` and `contentcache` packages could be either of these cases:

1. A look-aside cache. The `fs` package should be able to use the `gcsx` package to download the content from GCS API, and save it via the `contentcache` package.
2. A write-through cache. The `fs` package only talks to the `contentcache` package, which then uses the `gcsx` package to communicate with the APIs.

In either case, the responsibility of `gcsx` should be restricted to communications with the GCS APIs.

Reply:
I agree.  I will add a task to refactor this in a later PR but I think this change can go in for now.